core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -369,17 +310,20 @@ private UpdateResultHolder executeUpdateRequest(UpdateRequest updateRequest, Ind
         Translog.Location location = null;
         for (int i = 0; i < request.items().length; i++) {
             BulkItemRequest item = request.items()[i];
-            if (item.isIgnoreOnReplica() == false) {
+            assert item.getPrimaryResponse() != null;
```

Comment:
can we add a string telling use which item it is? the DocWriteRequest's toString is maybe an easy one